File: app/src/main/java/com/example/myapplication/activities/animepage/anime_detail_mvp/AnimeDetailInterface.java
package com.example.myapplication.activities.animepage.anime_detail_mvp;

import com.example.myapplication.models.mangamodels.DetailMangaModel;

import java.util.List;

public interface AnimeDetailInterface {
    void onGetDetailDataSuccess(DetailMangaModel detailMangaModel);

    void onGetDetailDataFailed();

    void onGetAllEpisodeSuccess(List<DetailMangaModel.DetailAllChapterDatas> detailAllEpisodeDataList);

    void onGetAllEpisodeFailed();

    void onGetGenreSuccess(List<DetailMangaModel.DetailMangaGenres> genresList);

    void onGetGenreFailed();
}

File: app/src/main/java/com/example/myapplication/activities/animepage/anime_detail_mvp/AnimeDetailInterface.kt
package com.example.myapplication.activities.animepage.anime_detail_mvp

import com.example.myapplication.models.mangamodels.DetailMangaModel

interface AnimeDetailInterface {
    fun onGetDetailDataSuccess(detailMangaModel: DetailMangaModel)

    fun onGetDetailDataFailed()

    fun onGetAllEpisodeSuccess(detailAllEpisodeDataList: List<DetailMangaModel.DetailAllChapterDatas>)

    fun onGetAllEpisodeFailed()

    fun onGetGenreSuccess(genresList: List<DetailMangaModel.DetailMangaGenres>)

    fun onGetGenreFailed()
}
